chains/evm/listener/event-handler.go

```diff
@@ -112,35 +133,49 @@ func Erc20EventHandler(sourceID, destId uint8, nonce uint64, handlerContractAddr
+	type GenericHandlerDepositRecord struct {
+		DestinationChainID uint8
```

Comment:
should be `DestinationDomainID`; we deprecated the use of `ChainID` in [PR 86](https://github.com/ChainSafe/chainbridge-core/pull/86) to avoid confusion with the `chainID` that go-ethereum uses

https://chainlist.org/

Reply:
and with this change, the ABI you use above should also be updated and pull from the newly generated ABI found within [consts package](https://github.com/ChainSafe/chainbridge-core/blob/main/chains/evm/calls/consts/consts.go) so as to remove the instances of `chainID`

Reply:
By and large these handlers will also undergo a refactor in the near future once [this PR](https://github.com/ChainSafe/chainbridge-core/pull/90) is completed and merged; this will remove the need for deposit record (and the call to fetch this data) by storing the deposit data within an event that gets emitted upon successful deposit; this all in effort to increase gas optimization at the contract level

Reply:
ok, thats really we should handle some how. @waymobetta when you think your PR will be ready?

Reply:
ok i guess its ok for now that they doing it with old fashioned way, when @waymobetta  PR will be merged additional fix to remove deposit record will be required that is additional work probably, but will dramaticly increase awareness of the project, so this problem could be skipped for now

Reply:
As I understand it now we have old contracts on master and new handler contract on develop in which deposit records mapping and  `getDepositRecord` is removed and  we fetch deposits from event on bridge contract?
Which contract tag are we targeting currently specifically?

So we should do it the `old` way still with `getDepositRecord`?

Reply:
yeap any way the new way of handling  is in progress